pkg/indicator/v2/bst/bst.go

```diff
@@ -0,0 +1,94 @@
+package bst
+
+// BST node.
+type Node struct {
+	value interface{}
+	left  *Node
+	right *Node
+}
+
+// BST type.
+type Tree struct {
+	root *Node
```

Comment:
maybe you can add a field for the type Comparable, and this Comparator could be passed from the caller

Reply:
Can you give a pseudo example what you would like to see?
Besides I agree the PR is getting too big but all code is tested and working state I would appreciate we merge having addressed almost all comments